crypto/evp_extra/p_dh.c

```diff
@@ -121,6 +122,29 @@ static int pkey_dh_ctrl(EVP_PKEY_CTX *ctx, int type, int p1, void *p2) {
   }
 }
 
+static int pkey_dh_ctrl_str(EVP_PKEY_CTX *ctx, const char *type,
+                            const char *value) {
+  // We don't support:
+  // * dh_paramgen_prime_len
+  // * dh_rfc5114
+  // * dh_param
+  // * dh_paramgen_generator
```

Comment:
Not for this PR, but `dh_paramgen_prime_len` and `dh_paramgen_generator` may be needed for Ruby :( ...
https://github.com/ruby/ruby/blob/ruby_3_1/ext/openssl/lib/openssl/pkey.rb#L120-L122